.bin/esyBuildCommand.js

```diff
@@ -0,0 +1,323 @@
+
+  let prelude = [
+    {
+      type: 'define',
+      name: 'ESY__SANDBOX',
+      value: '$(PWD)',
+      assignment: '?=',
+    },
+  ];
+
+  env.forEach(pkg => {
+    let info = pkg.packageJson;
+    let dependencies = collectTransitiveDependencies(envMap, info);
+
+    let findlibPath = dependencies
+      .map(dep => installDir(dep, 'lib'))
+      .join(':');
+
+    prelude.push({
+      type: 'define',
+      multiline: true,
+      name: `${info.name}__FINDLIB_CONF`,
+      value: `
+path = "$(shell ocamlfind printconf path):${findlibPath}"
+destdir = "${installDir(info.name, 'lib')}"
+      `.trim(),
+    });
+
+    rules.push({
+      type: 'rule',
+      name: null,
+      target: `${info.name}__findlib.conf`,
+      dependencies: [buildDir(info.name, 'findlib.conf')],
+      env: {},
+      command: null,
+    });
+
+    rules.push({
+      type: 'rule',
+      name: null,
+      target: buildDir(info.name, 'findlib.conf'),
+      dependencies: [],
+      env: {},
+      export: ['ESY__SANDBOX', `${info.name}__FINDLIB_CONF`],
+      command: `
+mkdir -p $(@D)
+echo "$${info.name}__FINDLIB_CONF" > $(@);
+      `.trim(),
+    });
+
+    if (info.pjc && info.pjc.build) {
+      let dependencies = [
+        `${info.name}__findlib.conf`
+      ];
+      if (info.dependencies) {
+        dependencies = dependencies.concat(Object.keys(info.dependencies));
+      }
+      rules.push({
+        type: 'rule',
+        name: ` *** Build ${info.name} ***`,
+        target: info.name,
+        dependencies: dependencies,
+        export: ['ESY__SANDBOX'],
+        env: []
+          .concat(envToEnvList(builtIns))
+          .concat(getBuildEnv(envMap, pkg))
+          .concat(envToEnvList(pkg)),
+        command: info.pjc.build,
+      });
+    } else {
+      // TODO: Returning an empty rule. Is that really what we want here?
+      rules.push({
+        type: 'rule',
+        name: ` *** Build ${info.name} ***`,
+        target: info.name,
+        export: ['ESY__SANDBOX'],
+        dependencies: info.dependencies != null
+          ? Object.keys(info.dependencies)
+          : [],
+        command: null,
+      });
+    }
+  });
+
+  console.log(renderMake(prelude.concat(rules)));
+}
+
+function getPkgEnv(envMap, pkg, current = false) {
+  let prefix = current ? 'cur' : (pkg.packageJson.name + '_');
+  let info = pkg.packageJson;
+  return [
+    {
+      name: `${prefix}_name`,
+      value: info.name,
+    },
+    {
+      name: `${prefix}_version`,
+      value: info.version,
+    },
+    {
+      name: `${prefix}_root`,
+      value: pkg.root,
+    },
+    {
+      name: `${prefix}_depends`,
+      value: info.dependencies != null
+        ? `"${Object.keys(info.dependencies).join(' ')}"`
+        : null,
+    },
+    {
+      name: `${prefix}_target_dir`,
+      value: `$_build_tree/node_modules/${info.name}`,
+    },
+    {
+      name: `${prefix}_install`,
+      value: `$_install_tree/node_modules/${info.name}/lib`,
```

Comment:
Now, in the `pjc` docs, I've said that the following: `_install_tree`, `sandbox`, `_build_tree` should be prefixed with `pjc__` ( `pjc__install_tree`, `pjc__sandbox`, `pjc__build_tree`) which means we should make them be ( `esy__install_tree`, `esy__sandbox` (already done), `esy__build_tree`).